openadapt/adapters/ultralytics.py

```diff
@@ -135,8 +137,29 @@ def do_fastsam(
         logger.info(f"{annotation.path=}")
         segmented_image_path = Path(tmp_dir) / result_name
         segmented_image = Image.open(segmented_image_path)
-        os.remove(segmented_image_path)
-
+        # os.remove(segmented_image_path)
+
+        # Ensure the image is fully loaded before deletion
+        segmented_image.load()
+
+        # Attempt to delete the file with retries and delay
+        max_retries = 5
+        retry_delay = 0.1  # seconds
```

Comment:
How about `retry_delay_seconds = 0.1`?

Reply:
Also what do you think about making these keyword arguments to this function with default values?

Reply:
yes it's better that way, I'll update this.

Reply:
Once you commit and push a change addressing a comment, please reply to the comment with the commit hash, or at least indicate with a 👍 that it is completed